tests/modules/cmdintrospection.c

```diff
@@ -0,0 +1,142 @@
+#include "redismodule.h"
+
+#define UNUSED(V) ((void) V)
+
+int cmd_xadd(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+    UNUSED(argv);
+    UNUSED(argc);
+    RedisModule_ReplyWithSimpleString(ctx, "OK");
+    return REDISMODULE_OK;
+}
+
+int RedisModule_OnLoad(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+    REDISMODULE_NOT_USED(argv);
+    REDISMODULE_NOT_USED(argc);
+
+    int spec_id;
+
+    if (RedisModule_Init(ctx, "cmdintrospection", 1, REDISMODULE_APIVER_1)== REDISMODULE_ERR)
+        return REDISMODULE_ERR;
+
+    if (RedisModule_CreateCommand(ctx,"cmdintrospection.xadd",NULL,"",0,0,0) == REDISMODULE_ERR)
+        return REDISMODULE_ERR;
+    RedisModuleCommandProxy *xadd = RedisModule_GetCommandProxy(ctx,"cmdintrospection.xadd");
+
+    if (RedisModule_AddCommandKeySpec(xadd,"write",&spec_id) == REDISMODULE_ERR)
+        return REDISMODULE_ERR;
+    if (RedisModule_SetCommandKeySpecBeginSearchIndex(xadd,spec_id,1) == REDISMODULE_ERR)
+        return REDISMODULE_ERR;
+    if (RedisModule_SetCommandKeySpecFindKeysRange(xadd,spec_id,0,1,0) == REDISMODULE_ERR)
+        return REDISMODULE_ERR;
+
+    RedisModule_SetCommandArity(xadd, -5);
+    RedisModule_SetCommandSummary(xadd, "Appends a new entry to a stream");
+    RedisModule_SetCommandDebutVersion(xadd, "5.0.0");
```

Comment:
did we agree that this refers to the module version or the redis version? (needs to be clearly documented, and maybe the example should reflect it).

maybe this is another thing to mention in the top comment. not sure how useful it is for clients...

Reply:
we agreed it's the module version.. i'll add a comment in the test module